services/spar/src/Spar/Scim.hs

```diff
@@ -148,111 +150,129 @@ validateSCIMUser ScimTokenInfo{stiIdP} user = do
+            Left err -> throwError $ Scim.badRequest Scim.InvalidValue
               (Just ("displayName is not compliant: " <> Text.pack err))
 
     uref :: SAML.UserRef <- do
       msubjid <- SAML.opaqueNameID <$$>
-        validateNameOrExtId (SCIM.User.externalId user)
+        validateNameOrExtId (Scim.User.externalId user)
 
       case msubjid of
         Just subj -> do
             pure $ SAML.UserRef (idp ^. SAML.idpMetadata . SAML.edIssuer) subj
         Nothing -> throwError $
-            SCIM.badRequest SCIM.InvalidValue (Just "externalId is required for SAML users")
+            Scim.badRequest Scim.InvalidValue (Just "externalId is required for SAML users")
 
-    handl <- case parseHandle (SCIM.User.userName user) of
+    handl <- case parseHandle (Scim.User.userName user) of
       Just x -> pure x
       Nothing -> throwError $
-        SCIM.badRequest SCIM.InvalidValue (Just "userName is not compliant")
+        Scim.badRequest Scim.InvalidValue (Just "userName is not compliant")
+
+    -- See this function's documentation
+    unless (null (Scim.User.emails user)) $ throwError $
+      Scim.badRequest Scim.InvalidValue (Just "emails currently can not be set via SCIM")
+    unless (null (Scim.User.phoneNumbers user)) $ throwError $
+      Scim.badRequest Scim.InvalidValue (Just "phone numbers currently can not be set via SCIM")
 
     -- We check the name for validity, but only if it's present
-    mbName <- Name <$$> validateNameOrExtId (SCIM.User.displayName user)
+    mbName <- Name <$$> validateNameOrExtId (Scim.User.displayName user)
 
     -- NB: We assume that checking that the user does _not_ exist has
     -- already been done before -- the hscim library check does a 'get'
     -- before a 'create'
 
-    pure $ ValidSCIMUser user uref handl mbName
+    pure $ ValidScimUser user uref handl mbName
 
 -- | We only allow SCIM users that authenticate via SAML.  (This is by no means necessary, though.
 -- It can be relaxed to allow creating users with password authentication if that is a requirement.)
-createValidSCIMUser
-  :: forall m. (m ~ SCIM.SCIMHandler Spar)
-  => ValidSCIMUser -> m SCIM.Class.User.StoredUser
-createValidSCIMUser (ValidSCIMUser user uref handl mbName) = do
+createValidScimUser
+  :: forall m. (m ~ Scim.ScimHandler Spar)
+  => ValidScimUser -> m Scim.Class.User.StoredUser
+createValidScimUser (ValidScimUser user uref handl mbName) = do
     -- This UserId will be used both for scim user in spar and for brig.
     buid <- Id <$> liftIO UUID.nextRandom
     -- Create SCIM user here in spar.
-    storedUser <- lift $ toSCIMStoredUser buid user
+    storedUser <- lift $ toScimStoredUser buid user
     lift . wrapMonadClient $ Data.insertScimUser buid storedUser
     -- Create SAML user here in spar, which in turn creates a brig user.
     lift $ createUser_ buid uref mbName
-    -- Set user handle on brig (which can't be done during user creation).
```

Comment:
I resurrected "set user handle" because there's no good reason why we can't do it, it's just that it hasn't been implemented in `NewUser`.